fe/fe-core/src/main/java/com/starrocks/scheduler/MvTaskRunProcessor.java

```diff
@@ -0,0 +1,500 @@
+            insertSqlBuilder.append(materializedView.getName());
+            insertSqlBuilder.append(" partition(" + mvPartitionName + ") ");
+            insertSqlBuilder.append(AST2SQL.toString(queryStatement));
+            String insertIntoSql = insertSqlBuilder.toString();
+            InsertStmt insertStmt = ((InsertStmt) SqlParser.parse(insertIntoSql,
+                    ctx.getSessionVariable().getSqlMode()).get(0));
+            insertStmt.setOrigStmt(new OriginStatement(insertIntoSql, 0));
+            insertStmt.setSystem(true);
+            StmtExecutor executor = new StmtExecutor(ctx, insertStmt);
+            ctx.setExecutor(executor);
+            ctx.setThreadLocalInfo();
+            try {
+                executor.execute();
+            } catch (Exception e) {
+                throw new SemanticException("Refresh materialized view failed:" + insertIntoSql, e);
+            } finally {
+                if (executor != null) {
+                    auditAfterExec(context, executor.getParsedStmt(), executor.getQueryStatisticsForAuditLog());
+                } else {
+                    // executor can be null if we encounter analysis error.
+                    auditAfterExec(context, null, null);
+                }
+            }
+        }
+    }
+
+    private Map<String, String> getPartitionProperties(MaterializedView materializedView) {
+        Map<String, String> partitionProperties = new HashMap<>(4);
+        partitionProperties.put("replication_num",
+                String.valueOf(materializedView.getDefaultReplicationNum()));
+        partitionProperties.put("storage_medium", materializedView.getStorageMedium());
+        String storageCooldownTime =
+                materializedView.getTableProperty().getProperties().get("storage_cooldown_time");
+        if (storageCooldownTime != null) {
+            partitionProperties.put("storage_cooldown_time", storageCooldownTime);
+        }
+        return partitionProperties;
+    }
+
+    private DistributionDesc getDistributionDesc(MaterializedView materializedView) {
+        HashDistributionInfo hashDistributionInfo =
+                (HashDistributionInfo) materializedView.getDefaultDistributionInfo();
+        List<String> distColumnNames = new ArrayList<>();
+        for (Column distributionColumn : hashDistributionInfo.getDistributionColumns()) {
+            distColumnNames.add(distributionColumn.getName());
+        }
+        return new HashDistributionDesc(hashDistributionInfo.getBucketNum(), distColumnNames);
+    }
+
+    private void addPartition(Database database, MaterializedView materializedView, long basePartitionId,
+                              PartitionKeyDesc partitionKeyDesc, Map<String, String> partitionProperties,
+                              DistributionDesc distributionDesc) {
+        // todo need to design partition name?
+        String partitionName = "mv_" + GlobalStateMgr.getCurrentState().getNextId();
```

Comment:
better to use partitionName of standard data warehouse. It it prefix by p.
There are different naming methods for different granularities. You can refer to the logic of dynamic partitioning or batch partitioning.